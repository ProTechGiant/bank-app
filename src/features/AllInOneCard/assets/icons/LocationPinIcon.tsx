import Svg, { Path } from "react-native-svg";

import type { IconProps } from "@/assets/icons";

export default function LocationPinIcon({ width = 24, height = 24, color = "#FF523D" }: IconProps) {
  return (
    <Svg width={width} height={height} viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
      <Path
        d="M12.0016 16.1496C13.6016 14.8496 14.8099 13.5996 15.6266 12.3996C16.4432 11.1996 16.8516 10.0663 16.8516 8.99961C16.8516 8.09961 16.6892 7.32861 16.3646 6.68661C16.0392 6.04528 15.6306 5.52861 15.1386 5.13661C14.6472 4.74528 14.1182 4.45794 13.5516 4.27461C12.9849 4.09128 12.4682 3.99961 12.0016 3.99961C11.5349 3.99961 11.0182 4.09128 10.4516 4.27461C9.8849 4.45794 9.3559 4.74528 8.86456 5.13661C8.37256 5.52861 7.96423 6.04528 7.63956 6.68661C7.31423 7.32861 7.15156 8.09961 7.15156 8.99961C7.15156 10.0663 7.5599 11.1996 8.37656 12.3996C9.19323 13.5996 10.4016 14.8496 12.0016 16.1496ZM12.0016 18.6246C11.8349 18.6246 11.6682 18.5996 11.5016 18.5496C11.3349 18.4996 11.1849 18.4246 11.0516 18.3246C8.96823 16.6913 7.41423 15.1036 6.38956 13.5616C5.36423 12.0203 4.85156 10.4996 4.85156 8.99961C4.85156 7.79961 5.07256 6.74528 5.51456 5.83661C5.9559 4.92861 6.51423 4.17028 7.18956 3.56161C7.86423 2.95361 8.62656 2.49528 9.47656 2.18661C10.3266 1.87861 11.1682 1.72461 12.0016 1.72461C12.8349 1.72461 13.6766 1.87861 14.5266 2.18661C15.3766 2.49528 16.1389 2.95361 16.8136 3.56161C17.4889 4.17028 18.0476 4.92861 18.4896 5.83661C18.9309 6.74528 19.1516 7.79961 19.1516 8.99961C19.1516 10.4996 18.6392 12.0203 17.6146 13.5616C16.5892 15.1036 15.0349 16.6913 12.9516 18.3246C12.8349 18.4246 12.6889 18.4996 12.5136 18.5496C12.3389 18.5996 12.1682 18.6246 12.0016 18.6246ZM12.0016 10.9996C12.5849 10.9996 13.0889 10.7913 13.5136 10.3746C13.9389 9.95794 14.1516 9.44961 14.1516 8.84961C14.1516 8.26628 13.9389 7.76628 13.5136 7.34961C13.0889 6.93294 12.5849 6.72461 12.0016 6.72461C11.4182 6.72461 10.9142 6.93294 10.4896 7.34961C10.0642 7.76628 9.85156 8.26628 9.85156 8.84961C9.85156 9.44961 10.0642 9.95794 10.4896 10.3746C10.9142 10.7913 11.4182 10.9996 12.0016 10.9996ZM6.00156 22.2746C5.6849 22.2746 5.41423 22.1663 5.18956 21.9496C4.96423 21.7329 4.85156 21.4663 4.85156 21.1496C4.85156 20.8163 4.96423 20.5413 5.18956 20.3246C5.41423 20.1079 5.6849 19.9996 6.00156 19.9996H18.0016C18.3182 19.9996 18.5889 20.1079 18.8136 20.3246C19.0389 20.5413 19.1516 20.8163 19.1516 21.1496C19.1516 21.4663 19.0389 21.7329 18.8136 21.9496C18.5889 22.1663 18.3182 22.2746 18.0016 22.2746H6.00156Z"
        fill={color}
      />
    </Svg>
  );
}
