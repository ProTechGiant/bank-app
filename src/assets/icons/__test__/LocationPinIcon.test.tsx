import { create } from "react-test-renderer";

import { LocationPinIcon } from "../LocationPinIcon";

describe("icon/LocationPinIcon", () => {
  it("renders a LocationPinIcon", () => {
    const result = create(<LocationPinIcon />);

    expect(result).toMatchSnapshot();
  });
});
