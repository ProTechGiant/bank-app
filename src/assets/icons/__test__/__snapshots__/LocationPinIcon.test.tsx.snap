// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon/LocationPinIcon renders a LocationPinIcon 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight="24"
  bbWidth="24"
  fill="none"
  focusable={false}
  height={24}
  meetOrSlice={0}
  minX={0}
  minY={0}
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "flex": 0,
        "height": 24,
        "width": 24,
      },
    ]
  }
  vbHeight={24}
  vbWidth={24}
  width={24}
  xmlns="http://www.w3.org/2000/svg"
>
  <RNSVGGroup
    fill={null}
    propList={
      [
        "fill",
      ]
    }
  >
    <RNSVGPath
      d="M18 8c0-3.31-2.69-6-6-6S6 4.69 6 8c0 4.5 6 11 6 11s6-6.5 6-11zm-8 0c0-1.1.9-2 2-2s2 .9 2 2a2 2 0 11-4 0zM5 20v2h14v-2H5z"
      fill={
        {
          "payload": 4278718035,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
        ]
      }
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;
